internal/provider/resources/flow.go

```diff
@@ -259,8 +250,63 @@ func (r *FlowResource) Read(ctx context.Context, req resource.ReadRequest, resp
 }
 
 // Update updates the resource and sets the updated Terraform state on success.
-func (r *FlowResource) Update(_ context.Context, _ resource.UpdateRequest, _ *resource.UpdateResponse) {
-	// Unsupported - redeploy by default
+func (r *FlowResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var plan FlowResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client, err := r.client.Flows(plan.AccountID.ValueUUID(), plan.WorkspaceID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.Append(helpers.CreateClientErrorDiagnostic("Flow", err))
+
+		return
+	}
+
+	var tags []string
+	resp.Diagnostics.Append(plan.Tags.ElementsAs(ctx, &tags, false)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	flowID, err := uuid.Parse(plan.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddAttributeError(
+			path.Root("id"),
+			"Error parsing Flow ID",
+			fmt.Sprintf("Could not parse flow ID to UUID, unexpected error: %s", err.Error()),
+		)
+
+		return
+	}
+
+	err = client.Update(ctx, flowID, api.FlowUpdate{
+		Tags: tags,
+	})
+	if err != nil {
+		resp.Diagnostics.Append(helpers.ResourceClientErrorDiagnostic("Flow", "update", err))
+
+		return
+	}
+
+	flow, err := client.Get(ctx, flowID)
```

Comment:
We need to `Get` the flow after updating it, because `Update` doesn't return the flow object.